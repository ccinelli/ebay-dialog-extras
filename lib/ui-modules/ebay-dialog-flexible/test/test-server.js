const testUtils = require('../../../helpers/server/test-utils');
const component = require('../');

testUtils.enableSnapshots();

// tests mirrored from eBayUI 

describe('ebay-dialog-flexible', () => {
    it('renders basic version', done => {
        testUtils.testSnapshot(component, {
            dontUsePortal: true,
        }, done);
    });
    it('renders with a renderBody', done => {
        testUtils.testSnapshot(component, {
            dontUsePortal: true,
            renderBody: out => out.write('Hello World') 
        }, done);
    });
    it('renders with a11y close text', done => {
        testUtils.testSnapshot(component, {
            dontUsePortal: true,
            a11yCloseText: 'Close Dialog' 
        }, done);
    });
    it('renders in open state', done => {
        testUtils.testSnapshot(component, {
            dontUsePortal: true,
            open: true 
        }, done);
    });
    [undefined, 'flexible', 'full'].forEach(type => {
        it(`renders with ${type || 'default'} type`, done => {
            const input = { type: type, dontUsePortal: true };
            testUtils.testSnapshot(component, input, done);
        });
    });
    it('renders with portal', done => {
        testUtils.testSnapshot(component, {
        }, done);
    });
});
