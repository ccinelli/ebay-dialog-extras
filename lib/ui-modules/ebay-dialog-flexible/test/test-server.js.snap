// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog renders basic version 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders in open state 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders with a renderBody 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\">Hello World</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders with a11y close text 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" aria-label=\\"Close Dialog\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders with default type 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders with flexible type 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--mask-fade\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`dialog renders with full type 1`] = `
"<div id=\\"w0\\" data-w-body=\\"0\\">
  <div class=\\"bolt-dialog dialog dialog--no-mask\\" hidden role=\\"dialog\\" id=\\"w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-onclick=\\"handleDialogClick|w0-thewidget\\" data-w-preserve-attrs=\\"hidden\\" data-w-events=\\"w0,dialog-show,handleShow,dialog-close,handleClose\\"
    data-w-body=\\"body\\">
    <div class=\\"dialog__window dialog__window--full dialog__window--fade\\" role=\\"document\\" id=\\"w0-thewidget-window\\">
      <button class=\\"dialog__close\\" type=\\"button\\" id=\\"w0-thewidget-close\\" data-w-onclick=\\"handleCloseButtonClick|w0-thewidget\\">
        <svg class=\\"icon icon--close\\" focusable=\\"false\\" id=\\"w0-thewidget-w0\\" aria-hidden=\\"true\\">
          <defs id=\\"w0-thewidget-w0-defs\\">
            <symbol id=\\"icon-close\\" viewBox=\\"1.5 1.5 21 21\\">
              <path d=\\"M22.182 3.856c.522-.554.306-1.394-.234-1.938-.54-.543-1.433-.523-1.826-.135C19.73 2.17 11.955 10 11.955 10S4.225 2.154 3.79 1.783c-.438-.371-1.277-.4-1.81.135-.533.537-.628 1.513-.25 1.938.377.424 8.166 8.218 8.166 8.218s-7.85 7.864-8.166 8.219c-.317.354-.34 1.335.25 1.805.59.47 1.24.455 1.81 0 .568-.456 8.166-7.951 8.166-7.951l8.167 7.86c.747.72 1.504.563 1.96.09.456-.471.609-1.268.1-1.804-.508-.537-8.167-8.219-8.167-8.219s7.645-7.665 8.167-8.218z\\"></path>
            </symbol>
          </defs>
          <use xlink:href=\\"#icon-close\\"></use>
        </svg>
      </button>
      <div class=\\"dialog__body\\" id=\\"w0-thewidget-body\\">
        <div class=\\"ebay-dialog-content-wrapper\\" id=\\"w0-0\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
